shotover-proxy/example-configs/cassandra-protect-local/topology.yaml

```diff
@@ -8,39 +8,7 @@ chain_config:
     - Protect:
         key_manager:
           Local:
-            kek:
-              - 0x1e
-              - 0xdf
-              - 0x0c
-              - 0xd6
-              - 0x70
-              - 0xce
-              - 0xff
-              - 0xb7
-              - 0xda
-              - 0xcb
-              - 0xe7
-              - 0x1f
-              - 0xb7
-              - 0xbd
-              - 0x4c
-              - 0xd9
-              - 0x7c
-              - 0xbb
-              - 0x8f
-              - 0x7d
-              - 0x04
-              - 0x9c
-              - 0xb0
-              - 0x2c
-              - 0x03
-              - 0xce
-              - 0x21
-              - 0x49
-              - 0x43
-              - 0x02
-              - 0x9b
-              - 0x59
+            kek: 0x1e0xdf0x0c0xd60x700xce0xff0xb70xda0xcb0xe70x1f0xb70xbd0x4c0xd90x7c0xbb0x8f0x7d0x040x9c0xb00x2c0x030xce0x210x490x430x020x9b0x59
```

Comment:
Is this some kind of standard I dont know about?
Wouldnt this be a lot nicer if each byte was exactly two characters and without the 'x's?

Reply:
It's the same format as before

Reply:
ok ... well no need to stick to it if we can come up with something better.

Reply:
I assumed that you get the key from AWS in that format.

Reply:
> I assumed that you get the key from AWS in that format, I'll see if I can find out what format it is and then we can accept that.

The current format isnt based on anything, its literally just deserializing directly into a vec of bytes.
So I agree, investigating what format these bytes actually come in is a great way forward!

> That said it's not uncommon for hex to have the leading 0x
Leading `0x` is good!
I wouldnt want that removed.

Reply:
>The current format isnt based on anything

Ah that makes sense, I assumed someone had made a demo key in AWS to use. They might even come in a different format (such as Base64) so it would be good to know.

Reply:
After a bit of research I think if we can make it accept hex and base64, that's pretty much all bases covered.

Reply:
I would write out a design that we can go over before you get too stuck into implementation problems. Could save you some time.

Reply:
> you get the key from AWS in that format

With KMS, you're typically never exposed to the key material itself, you simply ask AWS to encrypt/decrypt things with the key on your behalf. There is the option of uploading your own key material, but I've never seen the latter used. Re: format, AWS just lets you specify the path to the (encrypted) binary file. I was expecting the same for us - path to key file, rather than embedding the key in the config. Although our approach is more convenient. :)

> accept hex and base64, that's pretty much all bases covered

Yeah, that's a reasonable combination. Although base64 is standard, but the hex format arguably is not? What tool do we use to generate these keys, and how do we get it into the expected format? Should each byte be prefixed with `0x`, or is it enough to have only one at the start? Do we specify the format with another config, or is it auto-detected?